netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -351,22 +351,20 @@ public void run() {
     }
 
 
-    if (transportTracer != null) {
-      assert encoder().connection().equals(decoder().connection());
-      final Http2Connection connection = encoder().connection();
-      transportTracer.setFlowControlWindowReader(new TransportTracer.FlowControlReader() {
-        private final Http2FlowController local = connection.local().flowController();
-        private final Http2FlowController remote = connection.remote().flowController();
+    assert encoder().connection().equals(decoder().connection());
+    final Http2Connection connection = encoder().connection();
+    transportTracer.setFlowControlWindowReader(new TransportTracer.FlowControlReader() {
```

Comment:
it is recommended that you put this in a named class local to this method.  It makes fort better stack trace messages, rather than $3 or other hard to read names.  It usually only costs 1 line too.

Reply:
moved to a class in `Util` so that it can be reused in `NettyClientHandler`